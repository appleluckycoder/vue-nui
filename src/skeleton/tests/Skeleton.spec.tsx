import { mount } from '@vue/test-utils'
import { NSkeleton } from '../index'

describe('n-skeleton', () => {
  it('should work with import on demand', () => {
    mount(NSkeleton)
  })

  it('should work with `text` prop', async () => {
    const wrapper = mount(NSkeleton, {
      props: {
        text: true
      }
    })
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()
  })

  it('should work with box', async () => {
    const wrapper = mount(NSkeleton)
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()
  })

  it('should work with `sharp` prop', async () => {
    const wrapper = mount(NSkeleton, {
      props: {
        sharp: false
      }
    })
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()
  })

  it('should work with `round` prop', async () => {
    const wrapper = mount(NSkeleton, {
      props: {
        round: true
      }
    })
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()
  })

  it('should work with `circle` prop', async () => {
    const wrapper = mount(NSkeleton, {
      props: {
        circle: true
      }
    })
    expect(wrapper.find('.n-skeleton').attributes('style')).toMatchSnapshot()
  })

  it('should work with `repeat` prop', async () => {
    const wrapper = mount(NSkeleton, {
      props: {
        repeat: 2
      }
    })
    expect(wrapper.findAll('.n-skeleton').length).toBe(2)
  })
})
