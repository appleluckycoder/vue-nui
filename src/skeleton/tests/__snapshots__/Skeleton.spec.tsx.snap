// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`n-skeleton should work with \`circle\` prop 1`] = `"border-radius: 50%; --bezier: cubic-bezier(.4, 0, .2, 1); --color-start: #eee; --color-end: #ddd;"`;

exports[`n-skeleton should work with \`round\` prop 1`] = `"border-radius: 4096px; --bezier: cubic-bezier(.4, 0, .2, 1); --color-start: #eee; --color-end: #ddd;"`;

exports[`n-skeleton should work with \`sharp\` prop 1`] = `"border-radius: 3px; --bezier: cubic-bezier(.4, 0, .2, 1); --color-start: #eee; --color-end: #ddd;"`;

exports[`n-skeleton should work with \`text\` prop 1`] = `"display: inline-block; vertical-align: -0.125em; --bezier: cubic-bezier(.4, 0, .2, 1); --color-start: #eee; --color-end: #ddd;"`;

exports[`n-skeleton should work with box 1`] = `"--bezier: cubic-bezier(.4, 0, .2, 1); --color-start: #eee; --color-end: #ddd;"`;
